js/bundle_test.go

```diff
@@ -50,31 +50,44 @@ import (
 
 const isWindows = runtime.GOOS == "windows"
 
+func getRuntimeState(tb testing.TB, logger *logrus.Logger, rtOpts *lib.RuntimeOptions) *lib.RuntimeState {
+	if logger == nil {
+		logger = testutils.NewLogger(tb)
+	}
+	if rtOpts == nil {
+		rtOpts = &lib.RuntimeOptions{}
+	}
+	reg := metrics.NewRegistry()
+	return &lib.RuntimeState{
+		Logger:         logger,
+		RuntimeOptions: *rtOpts,
+		Registry:       reg,
+		BuiltinMetrics: metrics.RegisterBuiltinMetrics(reg),
+	}
+}
+
 func getSimpleBundle(tb testing.TB, filename, data string, opts ...interface{}) (*Bundle, error) {
-	var (
-		fs                        = afero.NewMemMapFs()
-		rtOpts                    = lib.RuntimeOptions{}
-		logger logrus.FieldLogger = testutils.NewLogger(tb)
-	)
+	fs := afero.NewMemMapFs()
+	var rtOpts *lib.RuntimeOptions
+	var logger *logrus.Logger
 	for _, o := range opts {
 		switch opt := o.(type) {
 		case afero.Fs:
 			fs = opt
 		case lib.RuntimeOptions:
-			rtOpts = opt
-		case logrus.FieldLogger:
+			rtOpts = &opt
+		case *logrus.Logger:
```

Comment:
Not certain if this doesn't break something ... maybe this `opts` should have a `default` case :thinking:

Reply:
Resolved by https://github.com/grafana/k6/pull/2627/commits/fe9f70ab769b0caeb31b07b86d95dd2b372ef436